frontend/ui/uimanager.lua

```diff
@@ -628,13 +630,9 @@ function UIManager:setRefreshRate(rate, night_rate)
     logger.dbg("set screen full refresh rate", rate, night_rate)
 
     if G_reader_settings:isTrue("night_mode") then
-        if night_rate then
-            self.FULL_REFRESH_COUNT = night_rate
-        end
+        self.FULL_REFRESH_COUNT = night_rate or self.FULL_REFRESH_COUNT
```

Comment:
Is this really an optimisation? I'd say the new stuff would do more assignments than the previous:
```diff
- if test then assignment end
+ test and assignment or assignment
```
(also, when reading that, I would translate in my head the new into the previous reading :)

Reply:
More assignments are done, when rate or night_reate are nil. But anyway, isn't the code more compact and more readable so?

Reply:
More assignments are done, when rate or night_reate are nil. But anyway, isn't the code more compact and more readable so?

Reply:
I'm... not a fan of those constructs either, FWIW ;).